lib/ClangImporter/ImportType.cpp

```diff
@@ -2382,6 +2382,12 @@ ImportedType ClangImporter::Implementation::importFunctionParamsAndReturnType(
   bool allowNSUIntegerAsInt =
       shouldAllowNSUIntegerAsInt(isFromSystemModule, clangDecl);
 
+  // If this function uses 'auto' return type, try to deduce the actual type.
+  if (clangDecl->getReturnType()->isUndeducedType())
+    getClangSema().DeduceReturnType(
```

Comment:
I wonder if this could make some template instantiations more eager (as we already have the overly eager instantiation problem).